src/sinks/new_relic/model.rs

```diff
@@ -54,43 +39,107 @@ impl TryFrom<Vec<Event>> for MetricsApiModel {
+                        metric_data.insert(
+                            "interval.ms".to_owned(),
+                            Value::from(interval_ms.get() as i64),
+                        );
+                        (value, "count")
                     }
+                    (MetricValue::Counter { value }, MetricKind::Absolute) => (value, "gauge"),
+                    (MetricValue::Gauge { value }, _) => (value, "gauge"),
                     _ => {
-                        // Unrecognized metric type
+                        // Unsupported metric type
+                        num_unsupported_metric_type += 1;
+                        return None;
                     }
+                };
+
+                // Set name, type, value, timestamp, and attributes
+                metric_data.insert("name".to_owned(), Value::from(series.name.name));
+                metric_data.insert("type".to_owned(), Value::from(metric_type));
+                let value = NotNan::new(value).ok().or_else(|| {
+                    num_nan_value += 1;
+                    None
+                })?;
+                metric_data.insert("value".to_owned(), Value::from(value));
+                metric_data.insert(
+                    "timestamp".to_owned(),
+                    Value::from(
+                        data.time
+                            .timestamp
+                            .unwrap_or_else(|| DateTime::<Utc>::from(SystemTime::now()))
+                            .timestamp(),
+                    ),
+                );
+                if let Some(tags) = series.tags {
+                    metric_data.insert(
+                        "attributes".to_owned(),
+                        Value::from(
+                            tags.iter_single()
+                                .map(|(key, value)| (key.to_string(), Value::from(value)))
+                                .collect::<BTreeMap<_, _>>(),
+                        ),
+                    );
                 }
-            }
+
+                Some(metric_data)
+            })
+            .collect();
+
+        if num_non_metric_events > 0 {
+            emit!(ComponentEventsDropped::<INTENTIONAL> {
+                count: num_non_metric_events,
+                reason: "non-metric event"
+            });
+        }
+        if num_unsupported_metric_type > 0 {
+            emit!(ComponentEventsDropped::<INTENTIONAL> {
+                count: num_unsupported_metric_type,
+                reason: "unsupported metric type"
+            });
+        }
+        if num_nan_value > 0 {
+            emit!(ComponentEventsDropped::<UNINTENTIONAL> {
+                count: num_nan_value,
+                reason: "NaN value not supported"
+            });
+        }
+        if num_missing_interval > 0 {
+            emit!(ComponentEventsDropped::<UNINTENTIONAL> {
+                count: num_missing_interval,
+                reason: "incremental counter missing interval"
+            });
         }
```

Comment:
Here and elsewhere, I'm not a fan of emitting these events within a `TryFrom` implementation but it made the most sense given the existing implementation. I'm happy to consider alternatives if others have strong opinions.